pkg/netpol/diff/diff_test.go

```diff
@@ -75,14 +63,150 @@ func TestDiff(t *testing.T) {
 			connsDiff, err := diffAnalyzer.ConnDiffFromDirPaths(firstDirPath, secondDirPath)
 			require.Empty(t, err)
 			actualOutput, err := diffAnalyzer.ConnectivityDiffToString(connsDiff)
-			if entry.isErr {
-				require.Equal(t, err.Error(), entry.expectedOutputErr)
-			} else {
-				require.Empty(t, err)
-				expectedOutputStr, err := os.ReadFile(expectedOutputFilePath)
-				require.Empty(t, err)
-				require.Equal(t, actualOutput, string(expectedOutputStr))
-			}
+			require.Empty(t, err)
+			expectedOutputStr, err := os.ReadFile(expectedOutputFilePath)
+			require.Empty(t, err)
+			require.Equal(t, actualOutput, string(expectedOutputStr))
+		}
+	}
+}
+
+type testErrEntry struct {
+	name            string
+	dir1            string
+	dir2            string
+	errStr          string
+	isCaFatalErr    bool
+	isCaOtherErr    bool // not fatal
+	isFormattingErr bool
+	format          string
+}
+
+var caErrType = &connectionsAnalyzingError{}     // error returned from a func on the ConnlistAnalyzer object
+var formattingErrType = &resultFormattingError{} // error returned from getting/writing output format
+
+func TestDiffErrors(t *testing.T) {
+	// following tests will be run with stopOnError, testing err string and diff err type
```

Comment:
don't we want to test with and without `stopOnError`? (not only with this option enabled)

Reply:
updated